ctapipe/image/muon/processor.py

```diff
@@ -0,0 +1,257 @@
+"""
+High level muon analysis  (MuonProcessor Component)
+"""
+import numpy as np
+
+from ctapipe.containers import (
+    ArrayEventContainer,
+    MuonParametersContainer,
+    MuonTelescopeContainer,
+)
+from ctapipe.coordinates import TelescopeFrame
+from ctapipe.core import QualityQuery, TelescopeComponent
+from ctapipe.core.traits import FloatTelescopeParameter, List, Tuple, Unicode
+
+from .features import (
+    intensity_ratio_inside_ring,
+    mean_squared_error,
+    ring_completeness,
+    ring_containment,
+)
+from .intensity_fitter import MuonIntensityFitter
+from .ring_fitter import MuonRingFitter
+
+INVALID = MuonTelescopeContainer()
+INVALID_PARAMETERS = MuonParametersContainer()
+
+
+class ImageParameterQuery(QualityQuery):
+    """
+    For configuring quality checks performed on image parameters
+    before rings are fitted.
+    """
+
+    quality_criteria = List(
+        Tuple(Unicode(), Unicode()),
+        default_value=[
+            ("min_pixels", "p.morphology.n_pixels > 100"),
+            ("min_intensity", "p.hillas.intensity > 500"),
+        ],
+        help=QualityQuery.quality_criteria.help,
```

Comment:
I think you should define here in the help message what the arguments are that the quality_query gets passed (and we should do that for the other ones as well)